app/models/project.rb

```diff
@@ -33,4 +37,15 @@ def related_projects
   def tasks_url
     "#{github_url}/issues"
   end
+
+  def self.parse_git_url(url)
+    url.gsub(/^(((https|http|git)?:\/\/(www\.)?)|git@)github.com(:|\/)/i, '').gsub(/(\.git|\/)$/i, '')
+  end
+
+  private
+
+  def set_github_repo
+    self.github_repo = Project.parse_git_url(self.submitted_github_url).split('/').last
```

Comment:
Line is too long. [87/80]<br>Redundant `self` detected.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.